Audiobook admin search

EdwinGuzman opened the pull request:
Added a radio select option in the admin search to select between audiobook, ebook, or all media types.

Ties in with [circulation #907](https://github.com/NYPL-Simplified/circulation/pull/907) and [server-core #833](https://github.com/NYPL-Simplified/server_core/pull/833).

aslagle commented on src/interfaces.ts:
```diff
@@ -325,3 +325,9 @@ export interface LaneData {
   lanes: LaneData[];
 }
+
+export interface Media {
```
There's already a `MediaData` interface in here, used when getting all the media types from "/admin/media". Can you reuse that?

aslagle commented on src/components/CustomListEditor.tsx:
```diff
@@ -23,6 +28,7 @@ export interface CustomListEditorState {
   name: string;
   entries: CustomListEntryData[];
   collections: AdminCollectionData[];
+  media?: Media[];
```
If you were going to use `MediaData` here, I would suggest adding a prop that contains all the media types (make a request to get them from the server on mount). And then the state can contain a list of the currently active media types rather than all of them.

EdwinGuzman commented:
Updated. I talked to Leonard about the options and agreed to filter the options so that only `Audio` and `Book` will come up in the UI.

aslagle commented on src/components/CustomListEditor.tsx:
```diff
@@ -32,14 +40,19 @@ export default class CustomListEditor extends React.Component<CustomListEditorPr
       entries: (this.props.list && this.props.list.entries) || [],
-      collections: (this.props.list && this.props.list.collections) || []
+      collections: (this.props.list && this.props.list.collections) || [],
+      media: this.props.media || {},
```
I don't think you need `media` in the state, doesn't look like you use it.

aslagle commented on src/components/CustomListEditor.tsx:
```diff
@@ -32,14 +40,19 @@ export default class CustomListEditor extends React.Component<CustomListEditorPr
     this.reset = this.reset.bind(this);
     this.search = this.search.bind(this);
+    this.changeMedia = this.changeMedia.bind(this);
+    this.getMediaQuery = this.getMediaQuery.bind(this);
```
I don't think you need to bind `getMediaQuery` here, since it's called from other functions in the component and not used as a React prop. It doesn't hurt anything though.

EdwinGuzman replied:
I typically bind functions even if they're not used as a prop. But, makes sense to not include.

aslagle replied:
It's fine either way, just wanted to mention it.

aslagle commented:
It looks like this will save your media selection when switching lists, which seems fine to me. If that is what we want I'd like to have a test for it.

EdwinGuzman commented:
Hmm I actually don't know if that is something we want. Resetting it makes more sense to me. What do you think?

aslagle commented:
I could go either way. I can't remember if it resets your search term when you change lists - maybe do the same thing it does with the search term.

EdwinGuzman commented:
Updated. Added a test in `CustomListEditor-test` that checks the state and value of the text input as the list prop gets updated. At first I tried to do it in `CustomList-test` since it seemed more appropriate to test there when you select a new list (or go to a new route). But, that proved difficult because I couldn't access the `CustomListEditor` component as a rendered element. And when I could access it, it didn't seem like I was testing it correctly.

EdwinGuzman commented:
Updated circulation and server_core with that update.
